### PR TITLE
Add fields to `Branch Protection` endpoint

### LINKED ISSUE #2719: Add fields to get branch protection API to match GitHub APIs
Upon comparing the API response of the [get branch protection endpoint](https://docs.github.com/en/rest/branches/branch-protection#get-branch-protection) and the struct implemented here I noticed a few fields missing:
- `required_pull_request_reviews.url string`
- `required_signatures.enabled boolean`
- `required_signature.url string`
- `required_status_checks.contexts []string`
- `required_status_checks.contexts_url string`
- `required_status_checks.url string`
- `url string`

Not sure if the `url` types of fields are left out on purpose but there are some cases when they are implemented (eg. `enforce_admins.url`) so I included them.

### PR DESCRIPTION
**Pull request by yardenshoham**
Fixes #2719

This adds a few fields that were requested in #2719.

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -834,6 +834,11 @@ type Branch struct {
 	Protected *bool             `json:"protected,omitempty"`
 }
 
+type RequiredSignatures struct {
```
Comment (gmlewis) [inline github/repos.go:837]: Please add a godoc-style comment to this exported struct (as a complete sentence with punctuation).

File: github/repos.go [hunk 2]
```diff
@@ -834,6 +834,11 @@ type Branch struct {
...
+	Url     string `json:"url"`
```
Comment (gmlewis) [inline github/repos.go:838]:
```suggestion
	URL     *string `json:"url,omitempty"`
```

File: github/repos.go [hunk 3]
```diff
@@ -834,6 +834,11 @@ type Branch struct {
...
+	Enabled bool   `json:"enabled"`
```
Comment (gmlewis) [inline github/repos.go:839]:
```suggestion
	Enabled *bool   `json:"enabled,omitempty"`
```

File: github/repos.go [hunk 4]
```diff
@@ -847,6 +852,8 @@ type Protection struct {
 	BlockCreations                 *BlockCreations                 `json:"block_creations,omitempty"`
 	LockBranch                     *LockBranch                     `json:"lock_branch,omitempty"`
 	AllowForkSyncing               *AllowForkSyncing               `json:"allow_fork_syncing,omitempty"`
+	RequiredSignatures             *RequiredSignatures             `json:"required_signatures,omitempty"`
+	Url                            *string                         `json:"url,omitempty"`
```
Comment (gmlewis) [inline github/repos.go:856]:
```suggestion
	URL                            *string                         `json:"url,omitempty"`
```

File: github/repos.go [hunk 5]
```diff
@@ -1015,6 +1022,8 @@ type ProtectionRequest struct {
 
 // RequiredStatusChecks represents the protection status of a individual branch.
 type RequiredStatusChecks struct {
+	Url         *string `json:"url,omitempty"`
```
Comment (gmlewis) [inline github/repos.go:1025]:
```suggestion
	URL         *string `json:"url,omitempty"`
```

File: github/repos.go [hunk 6]
```diff
@@ -1015,6 +1022,8 @@ type ProtectionRequest struct {
...
+	ContextsUrl *string `json:"contexts_url,omitempty"`
```
Comment (gmlewis) [inline github/repos.go:1026]:
```suggestion
	ContextsURL *string `json:"contexts_url,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @yardenshoham.
Please correct the issues below and make sure to re-run `gofmt` and `go generate ./...` and push (not force-push since we always squash and merge PRs in this repo) the changes to this PR.

**Comment by yardenshoham** [comment]
Thanks, changes applied

File: github/repos.go [hunk 7]
```diff
@@ -1015,6 +1017,8 @@ type ProtectionRequest struct {
 
 // RequiredStatusChecks represents the protection status of a individual branch.
 type RequiredStatusChecks struct {
+	URL         *string `json:"url,omitempty"`
```
Comment (gmlewis) [inline github/repos.go:1020]: Now that I'm looking at this a little more closely, let's please move these two new fields to the end of the struct and make `URL` the last field.  Thanks.

Comment (yardenshoham) [reply github/repos.go]: Done

**Review by gmlewis — APPROVED** [review]
Thank you, @yardenshoham !
LGTM.
Merging.

**Pull request merged**